src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/WindowsRuntime/Generated/WinRT/Projections.cs

```diff
@@ -0,0 +1,232 @@
+            }
+            finally
+            {
+                rwlock.ExitReadLock();
+            }
+        }
+
+        public static Type FindCustomPublicTypeForAbiType(Type abiType)
+        {
+            rwlock.EnterReadLock();
+            try
+            {
+                if (abiType.IsGenericType)
+                {
+                    return CustomAbiTypeToTypeMappings.TryGetValue(abiType.GetGenericTypeDefinition(), out Type publicTypeDefinition)
+                        ? publicTypeDefinition.MakeGenericType(abiType.GetGenericArguments())
+                        : null;
+                }
+                return CustomAbiTypeToTypeMappings.TryGetValue(abiType, out Type publicType) ? publicType : null;
+            }
+            finally
+            {
+                rwlock.ExitReadLock();
+            }
+        }
+
+        public static Type FindCustomTypeForAbiTypeName(string abiTypeName)
+        {
+            rwlock.EnterReadLock();
+            try
+            {
+                return CustomAbiTypeNameToTypeMappings.TryGetValue(abiTypeName, out Type type) ? type : null;
+            }
+            finally
+            {
+                rwlock.ExitReadLock();
+            }
+        }
+
+        public static string FindCustomAbiTypeNameForType(Type type)
+        {
+            rwlock.EnterReadLock();
+            try
+            {
+                return CustomTypeToAbiTypeNameMappings.TryGetValue(type, out string typeName) ? typeName : null;
+            }
+            finally
+            {
+                rwlock.ExitReadLock();
+            }
+        }
+
+        public static bool IsTypeWindowsRuntimeType(Type type)
+        {
+            Type typeToTest = type;
+            if (typeToTest.IsArray)
+            {
+                typeToTest = typeToTest.GetElementType();
+            }
+            return IsTypeWindowsRuntimeTypeNoArray(typeToTest);
+        }
+
+        private static bool IsTypeWindowsRuntimeTypeNoArray(Type type)
+        {
+            if (type.IsConstructedGenericType)
+            {
+                if(IsTypeWindowsRuntimeTypeNoArray(type.GetGenericTypeDefinition()))
+                {
+                    foreach (var arg in type.GetGenericArguments())
+                    {
+                        if (!IsTypeWindowsRuntimeTypeNoArray(arg))
+                        {
+                            return false;
+                        }
+                    }
+                    return true;
+                }
+                return false;
+            }
+            return CustomTypeToAbiTypeNameMappings.ContainsKey(type)
+                || type.IsPrimitive
+                || type == typeof(string)
+                || type == typeof(Guid)
+                || type == typeof(object)
+                || type.GetCustomAttribute<WindowsRuntimeTypeAttribute>() is object;
```

Comment:
Is the code `type.GetCustomAttribute<WindowsRuntimeTypeAttribute>() is object` here to determine whether this type has added WindowsRuntimeTypeAttribute

I think we can replace `type.IsDefined<WindowsRuntimeTypeAttribute>` to `type.GetCustomAttribute<WindowsRuntimeTypeAttribute>() is object` to improve performance